nym-vpn-desktop/src-tauri/src/commands/credential.rs

```diff
@@ -0,0 +1,53 @@
+use nym_vpn_lib::{credentials::import_credential_base58, nym_config::defaults::var_names};
+use tauri::{api::path::data_dir, State};
+use tracing::{debug, error, info, instrument};
+
+use crate::{
+    db::{Db, Key},
+    error::{CmdError, CmdErrorSource},
+    fs::util::check_dir,
+    APP_DIR,
+};
+
+const CREDENTIAL_DIR: &str = "credential";
+
+#[instrument(skip_all)]
+#[tauri::command]
+pub async fn add_credential(credential: String, db: State<'_, Db>) -> Result<(), CmdError> {
+    debug!("add_credential");
+    let network_name = std::env::var(var_names::NETWORK_NAME)
+        .inspect_err(|e| error!("failed to get current network name: {}", e))
+        .unwrap_or("unknown".into());
+    let mut path = data_dir().ok_or(CmdError::new(
+        CmdErrorSource::InternalError,
+        "Failed to retrieve data directory path".to_string(),
+    ))?;
+    path.push(APP_DIR);
+    path.push(CREDENTIAL_DIR);
```

Comment:
You can remove this line if you want to keep the same structure as other clients. If you want to keep it I'd reorder the credential dir and network name path push

Reply:
as discussed we'll use `_os_data_dir_/nym-vpn/backend/_network_name_` for cred dir